projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-results/search-ai-results.component.spec.ts

```diff
@@ -179,4 +149,22 @@ describe('SearchAiResultsComponent', () => {
       expect(getSkeletonElementsLength()).toBe(0);
     });
   });
+
+  describe('Unsaved Changes Modal', () => {
+    let modalAi: ModalAiService;
+
+    beforeEach(() => {
+      modalAi = TestBed.inject(ModalAiService);
+      spyOn(userPreferencesService, 'get').and.returnValue(knowledgeRetrievalNodes);
+    });
+
+    it('should open Unsaved Changes Modal', () => {
+      const modalAiSpy = spyOn(modalAi, 'openUnsavedChangesModal').and.callFake((callback) => callback());
```

Comment:
same like for previous testing file - we need to test what callback does because we need to ensure that logic inside callback works

Reply:
Here the same comment like above, please check